htlcswitch/switch.go

```diff
@@ -1103,9 +1103,15 @@ func (s *Switch) handlePacketForward(packet *htlcPacket) error {
 				failure = lnwire.NewTemporaryChannelFailure(update)
 			}
 
+			var nodeKey []byte
+			targetPeer := targetLink.Peer()
```

Comment:
It is already assumed on line 1041  that this `Peer` is available. I would just extract the node key there :)